sw/host/opentitanlib/src/io/i2c.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use thiserror::Error;
+use std::rc::Rc;
+use structopt::StructOpt;
+
+use crate::app::TransportWrapper;
+
+#[derive(Debug, StructOpt)]
+pub struct I2cParams {
+    #[structopt(long, help = "I2C instance", default_value = "0")]
+    pub bus: String,
+}
+
+impl I2cParams {
+    pub fn create(&self, transport: &TransportWrapper) -> Result<Rc<dyn Bus>> {
+        let i2c = transport.i2c(&self.bus)?;
+        Ok(i2c)
+    }
+}
+
+/// Errors related to the I2C interface and I2C transactions.
+#[derive(Error, Debug)]
+pub enum I2cError {
+    #[error("Invalid data length: {0}")]
+    InvalidDataLength(usize),
+}
+
+/// Represents a I2C transfer.
+pub enum Transfer<'rd, 'wr> {
+    Read(&'rd mut [u8]),
+    Write(&'wr [u8]),
+}
+
+/// A trait which represents a I2C Bus.
+pub trait Bus {
```

Comment:
For SPI, this has been called a "Target" and in you the hyperdebug i2c transport you have a struct "HyperdebugI2cTarget", so maybe this trait should be called "Target" to stay consistent?

Reply:
This was very intentional.  The spi trait includes the "chip select" pin, meaning that you are talking to a particular device on a particular SPI bus.  Hence, I assumed that "Target" in this context refers to a particular device connected to the bus. 
This I2C trait allows an "address" parameter, meaning that one single instance can be used to communicate with any device connected to the I2C bus.
Thanks for pointing out "HyperdebugI2cTarget".  That name was a mistake copied from SPI code.  I have renamed it to "HyperdebugI2cBus".

Reply:
Ah got it, sounds good thanks for clarifying.